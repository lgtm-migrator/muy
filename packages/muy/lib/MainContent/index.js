import React from "react"
import PropTypes from "prop-types"
import Box from "@material-ui/core/Box"

const MainContent = ({ id, mt, ...props }) => (
  <Box
    id={id}
    component="main"
    display="block"
    mt={mt}
    role="main"
    tabIndex={-1}
    {...props}
  />
)

export default MainContent

MainContent.defaultProps = {
  id: "content",
  mt: 3,
}

MainContent.propTypes = {
  id: PropTypes.string,
  mt: PropTypes.number,
}
