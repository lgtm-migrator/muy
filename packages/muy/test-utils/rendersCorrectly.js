import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
