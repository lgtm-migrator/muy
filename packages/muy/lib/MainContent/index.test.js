import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import MainContent from "."

describe("<MainContent/> component", () => {
  it("should render correctly with no props", () => {
    rendersCorrectly(<MainContent />)
  })
})
