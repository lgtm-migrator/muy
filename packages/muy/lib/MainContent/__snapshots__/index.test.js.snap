// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContent/> component should render correctly with no props 1`] = `
<main
  className="MuiBox-root MuiBox-root-2"
  id="content"
  role="main"
  tabIndex={-1}
/>
`;
